lib/modules/datasource/cpan/index.ts

```diff
@@ -0,0 +1,109 @@
+import { cache } from '../../../util/cache/package/decorator';
+import type { HttpResponse } from '../../../util/http/types';
+import * as perlVersioning from '../../versioning/perl';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { MetaCpanFileSearchResult } from './types';
+
+export class CpanDatasource extends Datasource {
+  static readonly id = 'cpan';
+
+  constructor() {
+    super(CpanDatasource.id);
+  }
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultRegistryUrls = ['https://fastapi.metacpan.org/'];
+
+  override readonly defaultVersioning = perlVersioning.id;
+
+  @cache({
+    namespace: `datasource-${CpanDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `${packageName}`,
+  })
+  override async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    let result: ReleaseResult | null = null;
+    const searchUrl = `${registryUrl}v1/file/_search`;
+
+    let raw: HttpResponse<MetaCpanFileSearchResult> | null = null;
+    try {
+      const body = {
+        query: {
+          filtered: {
+            query: { match_all: {} },
+            filter: {
+              and: [
+                { term: { 'module.name': packageName } },
+                { exists: { field: 'module.associated_pod' } },
+              ],
+            },
+          },
+        },
+        _source: [
+          'module.name',
+          'module.version',
+          'distribution',
+          'date',
+          'deprecated',
+          'maturity',
+        ],
+        sort: [{ date: 'desc' }],
+      };
+      raw = await this.http.postJson<MetaCpanFileSearchResult>(searchUrl, {
+        body,
+      });
+    } catch (err) {
+      this.handleGenericErrors(err);
+    }
+
+    const body = raw?.body;
+    if (body) {
+      const hits = body.hits.hits;
+      const releases: (Release & { distribution: string })[] = [];
```

Comment:
```suggestion
      const releases: (Release)[] = [];
```
remove that, it's not used by renovate

Reply:
Addressed in d045226.